carsons/carsons.py

```diff
@@ -65,6 +65,24 @@ def perform_kron_reduction(z_primitive: ndarray, dimension=3) -> ndarray:
     return Z_abc
 
 
+def calculate_self_impedance(Z):
```

Comment:
I'm wondering if these functions `calculate_self_impedance` and `calculate_mutual_impedance` are general enough -- i can imagine that depending on the use-case, a caller might think that self-impedance refers to just `z[i,i]` OR,  as you've defined it, the blended self-impedance that neglects phase-imbalance. likewise, the mutual may be thought of as the z[i,j] OR the blended average of all phase-phase mutuals. it might be better to just include these calculations directly where they are used -- its not as if they are especially hard to dream up either.

Reply:
Maybe, just removed them as they are redundant once Z012 is calculated